src/libtorch.cc

```diff
@@ -740,8 +740,11 @@ ModelInstanceState::ValidateInputs(const size_t expected_input_cnt)
       // input names since they are the keys for the dictionary
       input_index_map_[io_name] = i;
     } else {
-      // input tensor name must be in 'allowed_inputs' or must follow the naming
-      // convention
+      // Rules for (non-Dictionary) input tensor names:
+      // 1. Must be in 'allowed_inputs' (arguments in the forward function)
+      // 2. Must follow the naming convention i.e. <name>__<index>
+      // 3. If neither of the above conditions are satisfied, enforce strict
```

Comment:
Should this say "if both of the above conditions aren't satisfied"? Or some variation. Right now, it sounds like strict ordering will be followed only if 1 and 2 are both not true.

Reply:
> Right now, it sounds like strict ordering will be followed only if 1 and 2 are both not true.
It is a fall back if neither of the two conditions are met.